### PR TITLE
Add PhantomJs as an engine

### PR DESCRIPTION
**Pull request by nrother**
This code works, but is very basic, as no options are respected. Most of the options should be doable, since the rendering is completely controlled with the embedded javascript.

You can merge this if you like, or threat it as an RFC...

### CODE REVIEW

File: src/Pdf/Engine/PhantomJsEngine.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+use CakePdf\Pdf\CakePdf;
+use Cake\Core\Exception\Exception;
+
+class PhantomJsEngine extends AbstractPdfEngine
+{
+
+    /**
+     * Path to the phantomjs executable binary
+     *
+     * @access protected
```
Comment (dereuromark) [inline src/Pdf/Engine/PhantomJsEngine.php]: Those should be removed.

Comment (nrother) [inline src/Pdf/Engine/PhantomJsEngine.php]: @dereuromark The `@access protected`? Or the variable itself?

Comment (dereuromark) [inline src/Pdf/Engine/PhantomJsEngine.php]: `@access` is a relict from PHP4.

**Comment by dereuromark** [comment]
I think if this had some tests along with it this might be OK here.

**Comment by nrother** [comment]
The `@access` tags were removed. There are still some problems related with phantom js, somehow the page looks to large. One can add this to the CSS to fix this for the moment:
```css
html {
            zoom: 70%; /* HACK for phantomjs */
        }
```
But this is is [a bug in phantomjs](https://github.com/ariya/phantomjs/issues/12685).

File: src/Pdf/Engine/PhantomJsEngine.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    protected $_binary = '/usr/bin/phantomjs';
+    
+    /**
+     * Script used by PhantomJs for rendering the PDF
+     *
+     * @var string
+     */
+     protected $_phantomJsScript = <<<'EOT'
+     "use strict";
```
Comment (ceeram) [inline src/Pdf/Engine/PhantomJsEngine.php]: duplicate use strict

**Comment by ADmad** [comment]
Given that development of PhantomJS is [suspended](https://github.com/ariya/phantomjs/issues/15344) and we already have enough alternative engines closing this.

**Pull request closed without merging**